ui/components/search/search.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import querystring from 'querystring';
+
+import { redirectQuery } from '../lookup-search';
+
+export default class Search extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { searchTerm: '' };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ searchTerm: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { location } = this.props.router;
+    const query = redirectQuery(location.query, 'req_text__search', this.state.searchTerm);
+    const paramStr = querystring.stringify(query);
+    this.props.router.push(`/requirements/?${paramStr}`);
+  }
```

Comment:
I think we can avoid capturing this submission event in Javascript. Can we use an HTML form instead?
```
function hiddenFields(query) {
    const modifiedQuery = Object.assign({}, query);
    del modifiedQuery.page;
    del modifiedQuery.req_text__search;
    return Object.keys(modifiedQuery).map(k =>
        <input type="hidden" key={k} name={k} value={modifiedQuery[k]} />);
}

// will need to select actionPath from the current location
<form method="GET" action={ actionPath }>
    <input name="req_text__search" type="text" placeholder="Search..." />
    { hiddenFields(location.query) }
    <input type="submit" value="Submit" />
</form>
```

The benefit is that this works without JS.